functions/src/db/isTokenValidInDb.spec.ts

```diff
@@ -0,0 +1,73 @@
+// tslint:disable-next-line
+import '../testUtils/mockFirebaseAdminAndFunctions'
+import { JWTData } from '../types/JWT'
+import { firestoreGet } from '../testUtils/firestoreStub'
+import { isTokenValidInDb } from './isTokenValidInDb'
+
+describe('isTokenValidInDb', () => {
+    const token = 'tokenValue'
+    const jwtData: JWTData = {
+        uid: '69',
+        iat: 1544455,
+    }
+
+    it('should return true if everything is ok', async () => {
+        const snapshot = {
+            data: () => ({
+                tokens: {
+                    [token]: {
+                        valid: true,
+                    },
+                },
+            }),
+            exists: true,
+        }
+        firestoreGet.mockImplementation(() => Promise.resolve(snapshot))
+
+        const result = await isTokenValidInDb(jwtData, token)
+        expect(result).toBeTruthy()
+    })
+
+    it('should return false if nothing in db', async () => {
+        const snapshot = {
+            data: () => ({}),
+            exists: true,
+        }
+        firestoreGet.mockImplementation(() => Promise.resolve(snapshot))
+
+        const result = await isTokenValidInDb(jwtData, token)
+        expect(result).toBeFalsy()
+    })
```

Comment:
isn't this test case a more generic expression for `should return false if token is not existing in db` test ? I think this may be removed

Reply:
by bad, exists should have been set to false here.